docs/upgrading.md

```diff
@@ -5,6 +5,11 @@ the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summar
 
 ## Upgrading to v3.6
 
+### JSON templating fix
+
+When accessing a structure or array from json structure in a `jsonpath` expression you could end up getting a golang representation of the structure in error.
+This will now return the json representation as hoped - see #12909.
```

Comment:
```suggestion
This now returns plain JSON.
```

- simplify, per k8s style guide

I'm also not sure we need to reference the PR and I'm not sure that referencing it would be helpful. They sometimes are and aren't referenced in the upgrading notes, but when they are it is part of the section heading and not in the contents. A header reference makes more sense to me as a way to correlate to the changelog items.

Also the PR number is not automatically linked

Reply:
Done